lab_1_keywords_tfidf/main.py

```diff
@@ -66,7 +95,24 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if isinstance(frequencies, dict) is False or (isinstance(top, (int, float)) is False):
+        return None
+    else:
+        freq_val = list(frequencies.values())
+        words_num = 0
+        for value in freq_val:
+            words_num += value
+        print(words_num)
+        if top <= words_num:
+            """следующая абракадабра сортирует значения функций по убыванию(reverse = true, 
+             если этого не писать будет по возрастанию)
+             и выдает лист ключей, которым эти значения принадлежат"""
+            # нужно лучше разобраться в функции sorted
+            # get возвращает значение по указанному ключу
+            top_list = sorted(frequencies, key=frequencies.get, reverse=True)[:top]
+        else:
+            top_list = sorted(frequencies, key=frequencies.get, reverse=True)
+        print(top_list)
```

Comment:
consider using slices regardless of whether top is greater than dictionary volume 

```
print('abc'[:1000000000])
>> 'abc`
```